Riot/Modules/profile/ProfileViewController.swift

```diff
@@ -0,0 +1,210 @@
+                addMatrixSession(mxSession)
+            }
+        }
+        
+        signOutAlertPresenter.delegate = self
+        tableView.backgroundColor = theme.sideMenuProfileBackground
+        
+        //hiding the navigation bar's shadow
+        self.navigationController?.navigationBar.setValue(true, forKey: "hidesShadow")
+        self.navigationController?.navigationBar.backgroundColor = theme.sideMenuProfileBackground
+        view.backgroundColor = theme.sideMenuProfileBackground
+        self.navigationController?.navigationBar.backgroundColor = theme.sideMenuProfileBackground
+        
+        tableView.delegate = self
+        tableView.dataSource = self
+        
+        tableView.register(UINib(nibName: SectionType.PROFILE_CELL.cellIdentifier, bundle: nil), forCellReuseIdentifier: SectionType.PROFILE_CELL.cellIdentifier)
+        tableView.register(UINib(nibName: SectionType.ROLE_CELL.cellIdentifier, bundle: nil), forCellReuseIdentifier: SectionType.ROLE_CELL.cellIdentifier)
+        tableView.register(UINib(nibName: SectionType.ICON_ITEM_CELL.cellIdentifier, bundle: nil), forCellReuseIdentifier: SectionType.ICON_ITEM_CELL.cellIdentifier)
+        tableView.register(UINib(nibName: SectionType.ONLY_TEXT_CELL.cellIdentifier, bundle: nil), forCellReuseIdentifier: SectionType.ONLY_TEXT_CELL.cellIdentifier)
+        tableView.register(UINib(nibName: SectionType.ROLE_DIVIDER_CELL.cellIdentifier, bundle: nil), forCellReuseIdentifier: SectionType.ROLE_DIVIDER_CELL.cellIdentifier)
+        
+        roles.append(Role(name: "Software Developer", active: true))
+        roles.append(Role(name: "Business Analyst", active: false))
+        roles.append(Role(name: "UI Designer", active: false))
+        
+        iconItems.append(IconItem(image: UIImage(named: "settings_icon") ?? UIImage(), text: "Settings"))
+        iconItems.append(IconItem(image: UIImage(named: "role_outline") ?? UIImage(), text: "Roles"))
+        iconItems.append(IconItem(image: UIImage(named: "exit") ?? UIImage(), text: "Sign out"))
+        
+        textItems.append(TextItem(text: NSLocalizedString("settings_term_conditions", tableName: "Vector", bundle: Bundle.main, value: "", comment: ""), url: ""))
+        textItems.append(TextItem(text: NSLocalizedString("settings_privacy_policy", tableName: "Vector", bundle: Bundle.main, value: "", comment: ""), url: ""))
+        textItems.append(TextItem(text: "Acknowledgement", url: ""))
+        textItems.append(TextItem(text: "Version ... ", url: ""))
+    }
+    
+}
+
+extension ProfileViewController: UITableViewDataSource, UITableViewDelegate {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return SectionType.allCases.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case SectionType.ROLE_DIVIDER_CELL.rawValue, SectionType.ICON_ITEM_DIVIDER_CELL.rawValue:
+            return 1
```

Comment:
What does the number 1 represent here?  Can we either please create an appropriately named constant or add code comments

Reply:
This is a literal 'return 1 row for this section' value so we didn't think it appropriate to create a constant here.  We have added come comments here to explain why we are returning a single row for the divider and profile sections..